packages/cli/openapi-parser/src/asyncapi/parse.ts

```diff
@@ -229,22 +234,37 @@ function convertMessageToSchema({
     generatedName,
     event,
     context,
-    breadcrumbs
+    breadcrumbs,
+    options
 }: {
     breadcrumbs: string[];
     generatedName: string;
     event: AsyncAPIV2.PublishEvent | AsyncAPIV2.SubscribeEvent;
     context: AsyncAPIV2ParserContext;
+    options: ParseOpenAPIOptions;
 }): SchemaWithExample | undefined {
     if (event.message.oneOf != null) {
         const subtypes: (OpenAPIV3.SchemaObject | OpenAPIV3.ReferenceObject)[] = [];
+        const prefixes: (string | undefined)[] = [];
```

Comment:
would be simpler to type this as `string[]` and just prevent code paths from appending undefined values i _think_ otherwise we are at risk of a code path that goes `arr.join()` and having undefined show up in the name